universal-application-tool-0.0.1/conf/evolutions/default/3.sql

```diff
@@ -0,0 +1,31 @@
+# --- JSON-based tables
+
+# --- !Ups
+create table if not exists applicants (
+  id bigserial primary key,
+  object jsonb not null
+  -- No other constraint on applicant - empty object acceptable.
+);
+
+create table if not exists questions (
+  id bigserial primary key,
+  object jsonb not null,
+  check (object ? 'target')
+);
+
+create unique index if not exists question_target on questions(
+  (object ->>'target')
+);
```

Comment:
question: What does this do?

Reply:
Creates a lookup index - we can find the question by searching for the json-path that it gets stored in in the applicant json object.  Also ensures no two questions address the same piece of information.